### ISSUE TITLE
[java] UseConcurrentHashMap FN with assignment

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3962]
**Affects PMD Version:6.42.0** 

**Rule:**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.45.0/pmd_rules_java_multithreading.html#useconcurrenthashmap>

**Description:**
I found a false negative of this rule. I think PMD should report a warning  in line 3, but it doesn't.

**Code Sample demonstrating the issue:**
```java
public void m() {
    Map m;
    m = new HashMap(); // no warning
}
```
However, the following identical case can be detected.
```java
public void m() {
    Map m = new HashMap();  // can report a warning in this line
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
